src/services/AsyncStorageService.ts

```diff
@@ -1,26 +1,39 @@
 import AsyncStorage from '@react-native-async-storage/async-storage'
+import { OS_TYPES } from '../constants'
+import { Platform } from 'react-native'
 
 class AsyncStorageService {
   async getItem(key: string): Promise<unknown> {
     const jsonValue = await AsyncStorage.getItem(key)
-    const value = JSON.parse(jsonValue as string)
-    if (value) {
-      return value
+    try {
+      const value = JSON.parse(jsonValue as string)
+      if (value) {
+        return value
+      }
+    } catch (error) {
+      console.log('Error parsing JSON string: ', error)
+      if (jsonValue) {
+        return jsonValue
+      }
     }
 
     return null
   }
 
   async setItem(key: string, value: string): Promise<void> {
-    await AsyncStorage.setItem(key, JSON.stringify(value))
+    await AsyncStorage.setItem(key, value)
```

Comment:
Please, make sure to call stringify before the value is stored...
https://react-native-async-storage.github.io/async-storage/docs/usage#storing-object-value